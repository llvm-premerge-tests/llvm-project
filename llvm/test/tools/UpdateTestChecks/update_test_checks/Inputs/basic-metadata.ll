; Just run it through opt, no passes needed.
; RUN: opt < %s -S | FileCheck %s

define double @fpmathTest(double %a, double %b) {
entry:
  %ret = fmul double %a, %b, !fpmath !0
  ret double %ret
}

!0 = !{float 3.0}
