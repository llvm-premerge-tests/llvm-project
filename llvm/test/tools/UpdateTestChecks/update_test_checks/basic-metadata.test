## Basic test checking that update_test_checks.py differnt metadata
# RUN: cp -f %S/Inputs/basic-metadata.ll %t.ll && %update_test_checks %t.ll
# RUN: diff -u %t.ll %S/Inputs/basic-metadata.ll.expected
## Check that running the script again does not change the result:
# RUN: %update_test_checks %t.ll
# RUN: diff -u %t.ll %S/Inputs/basic-metadata.ll.expected
## Also try the --check-globals flag
# RUN: %update_test_checks %t.ll --check-globals
# RUN: diff -u %t.ll %S/Inputs/basic-metadata.ll.globals.expected
