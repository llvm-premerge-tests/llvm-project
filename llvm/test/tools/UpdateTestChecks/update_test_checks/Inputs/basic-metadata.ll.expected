; NOTE: Assertions have been autogenerated by utils/update_test_checks.py
; Just run it through opt, no passes needed.
; RUN: opt < %s -S | FileCheck %s

define double @fpmathTest(double %a, double %b) {
; CHECK-LABEL: @fpmathTest(
; CHECK-NEXT:  entry:
; CHECK-NEXT:    [[RET:%.*]] = fmul double [[A:%.*]], [[B:%.*]], !fpmath [[FPMATH0:![0-9]+]]
; CHECK-NEXT:    ret double [[RET]]
;
entry:
  %ret = fmul double %a, %b, !fpmath !0
  ret double %ret
}

!0 = !{float 3.0}
